* Disable database dump in preinst and database move in postinst
  if this is a unjoined univention system (Bug #28820)
* Adjust OpenLDAP version for database dumps (Bug #35667)
* do not dump if update43/slapd/database/dump is false (Bug #43645)

diff -Nur openldap-2.4.41+dfsg.o/debian/slapd.postinst openldap-2.4.41+dfsg/debian/slapd.postinst
--- openldap-2.4.41+dfsg.o/debian/slapd.postinst	2015-08-11 10:08:50.000000000 +0200
+++ openldap-2.4.41+dfsg/debian/slapd.postinst	2015-08-11 10:10:44.000000000 +0200
@@ -39,12 +39,18 @@ postinst_upgrade_configuration() {					# {{{
 		postinst_upgrade_cn_config
 	fi
 
-	# Check if the database format has changed.
-	if database_format_changed; then
+	local distributor="Debian"
+	if [ -x "/usr/sbin/ucr" ]; then
+		distributor="Univention"
+	fi
 
-		# During upgrading we have to load the old data
-		move_incompatible_databases_away
-		load_databases
+	if [ -e /var/univention-join/joined -o "Univention" != "$distributor" ]; then
+		# Check if the database format has changed.
+		if database_format_changed; then
+			# During upgrading we have to load the old data
+			move_incompatible_databases_away
+			load_databases
+		fi
 	fi
 
 	# Move to slapd.d configuration style.
diff -Nur openldap-2.4.42+dfsg.o/debian/slapd.scripts-common openldap-2.4.42+dfsg/debian/slapd.scripts-common
--- openldap-2.4.42+dfsg.o/debian/slapd.scripts-common	2017-03-08 00:30:48.244000000 +0100
+++ openldap-2.4.42+dfsg/debian/slapd.scripts-common	2017-03-08 00:32:20.524000000 +0100
@@ -26,6 +26,17 @@ database_dumping_enabled() {						# {{{
 	# don't know whether the next version will require reload.
 	[ "$MODE" = remove ] && return 0
 
+	local distributor="Debian"
+	if [ -x "/usr/sbin/ucr" ]; then
+		distributor="Univention"
+	fi
+	if [ ! -e /var/univention-join/joined -a "Univention" = "$distributor" ]; then
+		return 1
+	fi
+	if [ "xfalse" = "x$(ucr get update43/slapd/database/dump)" ]; then
+		return 1
+	fi
+
 	db_get slapd/dump_database
 	case "$RET" in
 	always)
